src/quo2/components/links/url_preview_list/view.cljs

```diff
@@ -0,0 +1,82 @@
+(ns quo2.components.links.url-preview-list.view
+  (:require
+    [oops.core :as oops]
+    [quo2.components.links.url-preview-list.style :as style]
+    [quo2.components.links.url-preview.view :as url-preview]
+    [react-native.core :as rn]
+    [reagent.core :as reagent]))
+
+(defn- use-scroll-to-last-item
+  [flat-list-ref item-count]
+  (rn/use-effect
+   (fn []
+     (when (pos? item-count)
+       ;; We use a delay because calling `scrollToIndex` without a delay does
+       ;; nothing while the flatlist is still rendering its children.
+       ;; `scrollToEnd` doesn't work because it positions the item off-center
+       ;; and there's no argument to offset it.
+       (let [timer-id (js/setTimeout
+                       (fn []
+                         (when (and @flat-list-ref (pos? item-count))
+                           (.scrollToIndex ^js @flat-list-ref
+                                           #js
+                                            {:index    (max 0 (dec item-count))
+                                             :animated true})))
+                       25)]
+         (fn []
+           (js/clearTimeout timer-id)))))
+   [item-count]))
+
+(defn- separator
+  []
+  [rn/view {:style style/url-preview-separator}])
+
+(defn- item-component
+  [{:keys [title body loading? logo]} _ _ {:keys [width on-clear loading-message]}]
+  [url-preview/view
+   {:logo            logo
+    :title           title
+    :body            body
+    :loading?        loading?
+    :loading-message loading-message
+    :on-clear        on-clear
+    :container-style {:width width}}])
+
+(defn- calculate-width
+  [preview-width horizontal-spacing ^js e]
+  (reset! preview-width
+    (- (oops/oget e "nativeEvent.layout.width")
+       (* 2 horizontal-spacing))))
+
+(defn- view-component
+  []
+  (let [preview-width (reagent/atom 0)
+        flat-list-ref (atom nil)]
+    (fn [{:keys [data key-fn horizontal-spacing on-clear loading-message]}]
+      (use-scroll-to-last-item flat-list-ref (count data))
+      ;; We need to use a wrapping view expanded to 100% instead of "flex 1",
+      ;; otherwise `on-layout` will be triggered multiple times as the flat list
+      ;; renders its children.
+      [rn/view
+       {:style               {:width "100%"}
+        :accessibility-label :url-preview-list}
+       [rn/flat-list
+        {:ref                               #(reset! flat-list-ref %)
+         :key-fn                            key-fn
+         :on-layout                         #(calculate-width preview-width horizontal-spacing %)
+         :horizontal                        true
+         :deceleration-rate                 :fast
+         :on-scroll-to-index-failed         identity
```

Comment:
Great question @ulisesmac. This might not be necessary, but I played safe here by copying what some other places do. See for instance this code comment screaming not to remove the workaround :laughing:  https://github.com/status-im/status-mobile/blob/develop/src/status_im2/contexts/chat/messages/list/view.cljs#L122 I can definitely remove it and if anything happens in the future we can fix it. Thanks for commenting on this.

In the Activity Center I once received feedback from QAs that after switching horizontally scrollable tabs, sometimes pressing on a tab would throw an exception. I never fully investigated why, but it certainly happened.

The documentation tells us this about `onScrollToIndexFailed`:

> Used to handle failures when scrolling to an index that has not been measured yet. Recommended action is to either compute your own offset and scrollTo it, or scroll as far as possible and then try again after more items have been rendered.

Now one thing is true, the `identity` function doesn't seem to be doing the recommended action at all. I wonder why the workaround even works... I'll remove this prop and see how it goes. Thanks!

Reply:
So @ulisesmac, I tried to change this and while experimenting with the component in another screen I got this error. It doesn't always happen, but I guess we'll need to keep the workaround.

<img src="https://user-images.githubusercontent.com/46027/231303479-131e16e6-1d89-4126-b64f-8aa96e9f8e5c.png" height="500" />

Reply:
Ohh ok! :(  thanks for taking a look at that property!  Yes, let's keep that workaround